src/main/java/lavalink/client/io/LavalinkRestClient.java

```diff
@@ -0,0 +1,182 @@
+                        callback.playlistLoaded(playlist);
+                        break;
+                    case "NO_MATCHES":
+                        callback.noMatches();
+                        break;
+                    case "LOAD_FAILED":
+                        final JSONObject exception = loadResult.getJSONObject("exception");
+                        final String message = exception.getString("message");
+                        final FriendlyException.Severity severity = FriendlyException.Severity.valueOf(exception.getString("severity"));
+                        final FriendlyException friendlyException = new FriendlyException(message, severity, new Throwable());
+                        callback.loadFailed(friendlyException);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Invalid loadType: " + loadType);
+                }
+            } catch (final Exception ex) {
+                callback.loadFailed(new FriendlyException(ex.getMessage(), FriendlyException.Severity.FAULT, ex));
+            }
+        };
+    }
+
+    private CompletableFuture<JSONObject> load(final String identifier) {
+        final CompletableFuture<JSONObject> future = new CompletableFuture<>();
+        executorService.submit(() -> {
+            try {
+                final String requestUrl = buildBaseAddress(socket) + URLEncoder.encode(identifier, "UTF-8");
+                final JSONObject response = apiGet(requestUrl, socket.getPassword());
+                future.complete(response);
+            } catch (final Throwable ex) {
+                future.completeExceptionally(ex);
+            }
+        });
+        return future;
+    }
+
+    private String buildBaseAddress(final LavalinkSocket socket) {
+        // wss:// or ws:// -> http://
+        return socket.getRemoteUri().toString().replaceFirst("[wW][sS]{1,2}[:][/]{2}", "http://") + "/loadtracks?identifier=";
+    }
+
+    private HttpClient buildClient() {
+        final HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        if (builderConsumer == null)
+            return httpClientBuilder.build();
+        builderConsumer.accept(httpClientBuilder);
+        return httpClientBuilder.build();
+    }
+
+    private JSONObject apiGet(final String url, final String auth) throws IOException {
+        final HttpGet request = new HttpGet(url);
+        request.setHeader(HttpHeaders.AUTHORIZATION, auth);
+        final HttpClient httpClient = buildClient();
+        final HttpResponse httpResponse = httpClient.execute(request);
+        final int statusCode = httpResponse.getStatusLine().getStatusCode();
+        if (statusCode != 200)
+            throw new IOException("Invalid API Request Status Code: " + statusCode);
+        final HttpEntity entity = httpResponse.getEntity();
+        if (entity == null)
+            throw new IOException("Invalid API Response: No Content");
```

Comment:
```suggestion
            throw new IOException("Empty API response");
```

You already check against No Content